### PR TITLE
fix: fall back to default value when `publicPath` is falsy

### LINKED ISSUE #4364: Validating publicPath value in Nuxt Config [v2.3.1]
### Version

[v2.3.1](https://github.com/nuxt.js/releases/tag/v2.3.1)

### Reproduction link

[https://codesandbox.io/s/pxxv9r660](https://codesandbox.io/s/pxxv9r660)

### Steps to reproduce

In the `build` section of the nuxt.config.js file, set the `publicPath` value to an empty string, or false. 

Then re-run `npm run dev`

### What is expected ?

Nuxt should default to `/_nuxt/` in the event of an invalid value.

### What is actually happening?

If you set it to an empty string, the app starts up as expected, but every page returns a 404 with no client or server side errors messages. 

If you set it to `false`, you receive this error:
`url.startsWith is not a function`

### Additional comments?

In my 2.2.0 build, I had this value set to an empty string for my local environment and an S3 bucket when live. 

This worked fine for the last few months. Upon updating, every page of my app 404'd with no explanation. 

I tore down the config line-by-line and discovered that the publicPath was causing the issue. 

It's fairly clearly stated in the docs that the default value should be `/_nuxt/`, but since an empty string worked in previous versions, other people who are as foolish as I am might run into this issue.

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8175">#c8175</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by TheAlexLichter**
## Types of changes
- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Resolves #4364 

## Checklist:
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [x] All new and existing tests are passing.

### CODE REVIEW

**Comment by manniL** [comment]
Would suggest to release a patch (details in the linked issue above) after merging.

**Review by pi0 — DISMISSED** [review]

**Comment by pi0** [comment]
@manniL I have an alternative way to fix this: We can force add a trailing slash to the publicPath. This not only resolves this bug but also auto-fixes the situations that user forgot to add it.

**Comment by manniL** [comment]
@pi0 I'd combine both solutions 👍

File: packages/config/src/options.js [hunk 1]
```diff
@@ -111,6 +118,10 @@ export function getNuxtConfig(_options) {
     options.appTemplatePath = path.resolve(options.srcDir, options.appTemplatePath)
   }
 
+  // Add trailing slash to path if not already present
+  if (!/\/$/.test(options.build.publicPath)) {
```
Comment (clarkdo) [inline packages/config/src/options.js]: Same for `_publicPath`?

Comment (clarkdo) [reply packages/config/src/options.js]:
```suggetsion
options.build.publicPath = options.build.publicPath.replace(/([^/])$/, '$1/')
options.build._publicPath= options.build._publicPath.replace(/([^/])$/, '$1/')
```

Comment (clarkdo) [inline packages/config/src/options.js]:
```suggestion
  options.build.publicPath = options.build.publicPath.replace(/([^/])$/, '$1/')
  options.build._publicPath= options.build._publicPath.replace(/([^/])$/, '$1/')
```

Comment (clarkdo) [reply packages/config/src/options.js]:
```suggestion
  options.build.publicPath = options.build.publicPath.replace(/([^/])$/, '$1/')
  options.build._publicPath= options.build._publicPath.replace(/([^/])$/, '$1/')
```

Comment (manniL) [reply packages/config/src/options.js]: Good catch!

**Review by clarkdo — APPROVED** [review]

**Pull request merged**